static/src/javascripts/projects/common/utils/robust.js

```diff
@@ -0,0 +1,35 @@
+/*jshint -W024 */
+define([
+    'raven',
+    'Promise'
+], function (
+    raven,
+    Promise
+) {
+
+    /*
+    * This purposefully does not return the Promise.
+    * It is designed to swallow and report on exceptions at a high level.
+    * If you need to act on the return value you are looking for a different abstraction
+    * */
+    function Robust(name, block, reporter) {
+
+        if (!reporter) {
+            reporter = raven.captureException;
+        }
+
+        new Promise(function (resolve, reject) {
+            try {
+                block();
+                resolve();
+            } catch (e) {
+                reject(e);
+            }
+        }).catch(function (e) {
+            reporter(e, { tags: { module: name } });
+        });
```

Comment:
Good call. I started out actually intending to use the promises, but realised they were not exactly what I wanted. It did feel to me like the page rendered smoother with them in so I left them, but I ran some numbers this morning using the timing api and I don't think that is actually the case. Will replace with simpler `try { ... } catch (e) { ... }`.